modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/IndexManager.java

```diff
@@ -281,6 +303,66 @@ boolean isIndexStale(ShpFileType indexType) {
     }
 
     /**
+     * Queries the spatial index for features available in the specified bbox, 
+     * applying the BBOX filter if specifided.
+     * 
+     * @param bbox
+     */
+    protected CloseableIterator<Data> querySpatialIndex(Envelope bbox, boolean applyEnvelopeFilter) throws DataSourceException, TreeException, IOException {
```

Comment:
Hi @aaime, I know your advices are right, but I do not know other mode to avoid that the BBOX defined in a part of mixed alpahumeric-spatial filter rejects features that dont share these envelope but matches the alphanumeric clause. I managed to explain my intentions?

Thanks Andrea for your patience.